src/lib/detection_algorithms/cntk/exporter.js

```diff
@@ -0,0 +1,158 @@
+                    cb();
+                },
+                ensureDirExists.bind(null, self.exportDirPath),
+                ensureDirExists.bind(null, self.posDirPath),
+                ensureDirExists.bind(null, self.negDirPath),
+                ensureDirExists.bind(null, self.testDirPath)
+            ], function(err) {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+                resolve();
+            });
+        });
+    }
+
+    // Export a single frame to the training data
+    // Parameters:
+    //  frameFileName - The file name to use when saving the image file 
+    //  frameBuffer - A buffer with the frame image data 
+    //  bboxes  - a list of bboxes in the format of x1, y1, x2, y2 where the
+    //           coordinates are in absolute values of the image
+    //  tags - a list of objects containing the tagging data. Each object is in the format of:
+    //         {'x1' : int, 'y1' : int, 'x2' : int, 'y2' : int, 'class' : string}
+    //         Where (x1,y1) and (x2,y2) are the coordinates of the top left and bottom right corner
+    //         of the bounding boxes (respectively), and 'class' is the name of the class.
+    // Returns: A Promise object that resolves when the operation completes
+    this.exportFrame = function exportFrame(frameFileName, frameBuffer, tags) {
+        return new Promise(function(resolve, reject) {
+            async.waterfall([
+                deleteFileIfExists.bind(null, path.join(self.negDirPath,  frameFileName)),
+                deleteFileIfExists.bind(null, path.join(self.posDirPath,  frameFileName)),
+                deleteFileIfExists.bind(null, path.join(self.testDirPath, frameFileName)),
+                deleteFileIfExists.bind(null, path.join(self.posDirPath,  `${frameFileName}.bboxes.labels.tsv`)),
+                deleteFileIfExists.bind(null, path.join(self.testDirPath, `${frameFileName}.bboxes.labels.tsv`)),
+                deleteFileIfExists.bind(null, path.join(self.posDirPath,  `${frameFileName}.bboxes.tsv`)),
+                deleteFileIfExists.bind(null, path.join(self.testDirPath, `${frameFileName}.bboxes.tsv`)),
+                function determineWritePath(cb){
+                    if( !tags.length){ 
+                        cb(null, self.negDirPath);
+                    } else {
+                         if (self.testFrameIndecies.includes(self.posFrameIndex)){
+                             self.posFrameIndex++;
+                             cb(null, self.testDirPath);
+                         } else {
+                             self.posFrameIndex++;
+                             cb(null, self.posDirPath);
+                         }
+                    }
+                },
+                function saveImage(exportPath, cb) {
+                    var imageFilePath  = path.join(exportPath, frameFileName);
+                    fs.writeFileSync(imageFilePath, frameBuffer);
```

Comment:
Not sure what is the question, but since you already receiving a callback as a parameter, you can just do:
```
fs.writeFile(imageFilePath, frameBuffer, cb)
```
Instead of lines 91+92